src/basic/unit-name.c

```diff
@@ -507,6 +509,42 @@ int unit_name_template(const char *f, char **ret) {
         return 0;
 }
 
+int unit_name_hash_long(const char *name, char **ret) {
+        _cleanup_free_ char *s = NULL;
+        char hash[17] = {}, n[UNIT_NAME_MAX] = {}, *suffix;
+        uint64_t h;
+        uint8_t k[16] = {};
+        int i;
+
+        if (strlen(name) < UNIT_NAME_MAX)
+                return 0;
+
+        suffix = strrchr(name, '.');
+        if (!suffix)
+                return -EINVAL;
+
+        if (unit_type_from_string(suffix+1) < 0)
+                return -EINVAL;
+
+        /* We intentionally hash with k that is all zeros because each time we want the same hash value for the same string. */
+        h = siphash24_string(name, k);
```

Comment:
Maybe define a static `k` declared using `SD_ID128_MAKE()`? It's done in https://github.com/systemd/systemd/blob/main/src/udev/udev-node.c#L256